packages/frontend/src/components/UnderlyingData/DrillDownMenuItem.tsx

```diff
@@ -0,0 +1,201 @@
+                    ? FilterOperator.NULL
+                    : FilterOperator.EQUALS,
+            values:
+                rowValue.value.raw === null ? undefined : [rowValue.value.raw],
+        };
+        return [...acc, dimensionFilter];
+    }, []);
+    combinedDimensionFilters.push(...dimensionFilters);
+
+    return {
+        dimensions: {
+            id: uuidv4(),
+            and: combinedDimensionFilters,
+        },
+    };
+};
+
+type DrillDownExploreUrlArgs = {
+    projectUuid: string;
+    tableName: string;
+    metricQuery: MetricQuery;
+    row: ResultRow;
+    drillByMetric: FieldId;
+    drillByDimension: FieldId;
+    dashboardFilters?: DashboardFilters;
+    extraFilters?: Filters;
+    pivotReference?: PivotReference;
+};
+
+const drillDownExploreUrl = ({
+    projectUuid,
+    tableName,
+    metricQuery,
+    row,
+    drillByMetric,
+    drillByDimension,
+    dashboardFilters,
+    extraFilters,
+    pivotReference,
+}: DrillDownExploreUrlArgs) => {
+    const createSavedChartVersion: CreateSavedChartVersion = {
+        tableName,
+        metricQuery: {
+            tableCalculations: [],
+            dimensions: [drillByDimension],
+            metrics: [drillByMetric],
+            filters: combineFilters({
+                metricQuery,
+                row,
+                dashboardFilters,
+                extraFilters,
+                pivotReference,
+            }),
+            limit: 500,
+            additionalMetrics: metricQuery.additionalMetrics,
+            sorts: [{ fieldId: drillByDimension, descending: false }],
+        },
+        pivotConfig: undefined,
+        tableConfig: {
+            columnOrder: [],
+        },
+        chartConfig: {
+            type: ChartType.CARTESIAN,
+            config: { layout: {}, eChartsConfig: {} },
+        },
+    };
+    const { pathname, search } = getExplorerUrlFromCreateSavedChartVersion(
+        projectUuid,
+        createSavedChartVersion,
+    );
+    return `${pathname}?${search}`;
+};
+
+export const DrillDownMenuItem: FC<{
+    metricQuery: MetricQuery | undefined;
+    row: ResultRow | undefined;
+    selectedItem: Field | TableCalculation | undefined;
+    dashboardFilters?: DashboardFilters;
+    pivotReference?: PivotReference;
+}> = ({ metricQuery, row, selectedItem, dashboardFilters, pivotReference }) => {
+    const { projectUuid } = useParams<{ projectUuid: string }>();
+    const { explore } = useUnderlyingDataContext();
+    if (
+        selectedItem &&
+        isField(selectedItem) &&
+        isMetric(selectedItem) &&
+        explore &&
+        row &&
+        metricQuery
+    ) {
+        return (
+            <MenuItem2 text="Drill by" icon="path">
+                {Object.values(explore.tables).map((table) => (
+                    <MenuItem2 key={table.name} text={table.label}>
+                        {Object.values(table.dimensions)
+                            .filter((dimension) => !dimension.hidden)
```

Comment:
It seems this hidden filter is not working..

Reply:
> I think we need to hide hidden fields like Order date

`Order date` Is not a hidden field. We just have some logic in the sidebar to transform it into a collapsible group.
There is a ticket ( can't find it ) about removing that option from the filters autocomplete as well.